File: benchmark/output/python-pygame/gemini/shooting.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Vibe Shooter")
clock = pygame.time.Clock()

# Player state
player_x = 384.0
player_y = 550.0
player_speed = 200.0

# Bullet state (Parallel lists)
bullet_xs = []
bullet_ys = []
bullet_speed = 300.0

# Track spacebar state to prevent rapid-fire on hold
space_pressed = False

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    keys = pygame.key.get_pressed()
    
    # Player movement
    if keys[pygame.K_LEFT] and player_x > 0:
        player_x -= player_speed * dt
    if keys[pygame.K_RIGHT] and player_x < 768:
        player_x += player_speed * dt

    # Shooting logic
    if keys[pygame.K_SPACE]:
        if not space_pressed:
            bullet_xs.append(player_x + 14) # Center of 32px rect
            bullet_ys.append(player_y)
            space_pressed = True
    else:
        space_pressed = False

    # Update bullets
    for i in range(len(bullet_ys)):
        bullet_ys[i] -= bullet_speed * dt

    # Draw
    screen.fill((0, 0, 0))
    
    # Draw player
    pygame.draw.rect(screen, (255, 255, 255), (int(player_x), int(player_y), 32, 32))
    
    # Draw bullets
    for i in range(len(bullet_xs)):
        pygame.draw.rect(screen, (255, 255, 255), (int(bullet_xs[i]), int(bullet_ys[i]), 4, 4))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/gemini/shooting.lua
local player_x = 384
local player_y = 550
local player_speed = 200
local bullet_speed = 300
local bullet_xs = {}
local bullet_ys = {}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Vibe Shooter")
end

function love.keypressed(k)
    if k == "space" then
        table.insert(bullet_xs, player_x + 14) -- Center bullet in 32px player
        table.insert(bullet_ys, player_y)
    end
end

function love.update(dt)
    -- Player movement
    if love.keyboard.isDown("left") and player_x > 0 then
        player_x = player_x - player_speed * dt
    end
    if love.keyboard.isDown("right") and player_x < 768 then
        player_x = player_x + player_speed * dt
    end

    -- Bullet movement
    for i = 1, #bullet_ys do
        bullet_ys[i] = bullet_ys[i] - bullet_speed * dt
    end
end

function love.draw()
    -- Draw player
    love.graphics.setColor(1, 1, 1)
    love.graphics.rectangle("fill", player_x, player_y, 32, 32)

    -- Draw bullets
    love.graphics.setColor(1, 0, 0)
    for i = 1, #bullet_xs do
        love.graphics.rectangle("fill", bullet_xs[i], bullet_ys[i], 4, 4)
    end
    
    -- Reset color
    love.graphics.setColor(1, 1, 1)
end